web-app/src/app/tabs/race/race-graph/race-graph.component.ts

```diff
@@ -0,0 +1,200 @@
+      });
+
+    this.chart = new CanvasJS.Chart('raceGraph', {
+      exportEnabled: true,
+      theme: 'dark1',
+      backgroundColor: 'transparent',
+      title: {
+        text : 'Distance Remaining',
+        fontSize: 20
+      },
+      data: [{
+        type: 'line',
+        dataPoints : this.distanceArray, // Start off by displaying lap distance graph
+      },
+      {
+        type: 'line',
+        dataPoints: [],
+      },
+      {
+        type: 'line',
+        dataPoints: [],
+      }],
+
+      axisX: {
+        title: 'Lap Number',
+        fontSize: 15,
+      },
+
+      axisY: {
+        title: 'Distance (m)',
+        fontSize: 15,
+      }
+    });
+
+  this.chart.render();
+  }
+
+  async initializeDataArrays() {
+    this.lapData = this.lapService.getData()
+    this.currLap = this.lapData.length;
+    let lapDataIdx =  this.lapData.length;
+
+    for (const dataPoint of this.lapData) {
+      this.lapTimeArray.unshift({x: lapDataIdx, y: this.convertToMinutes(dataPoint.lapTime)});
+      this.totalPowerInArray.unshift({x: lapDataIdx, y: dataPoint.totalPowerIn});
+      this.totalPowerOutArray.unshift({x: lapDataIdx, y: dataPoint.totalPowerOut});
+      this.netPowerOutArray.unshift({x: lapDataIdx, y: dataPoint.netPowerOut});
+      this.distanceArray.unshift({x: lapDataIdx, y: dataPoint.distance});
+      this.amphoursArray.push({x: lapDataIdx, y: dataPoint.amphours});
+      this.averagePackCurrentArray.unshift({x: lapDataIdx, y: dataPoint.averagePackCurrent});
+      this.batterySecondsRemainingArray.unshift({x: lapDataIdx, y: dataPoint.batterySecondsRemaining});
+      lapDataIdx--
+    }
+  }
+
+  addLapData(dataPoint: LapData) {
+      this.currLap++;
+      this.lapTimeArray.push({x: this.currLap, y: this.convertToMinutes(dataPoint.lapTime)});
+      this.totalPowerInArray.push({x: this.currLap, y: dataPoint.totalPowerIn});
+      this.totalPowerOutArray.push({x: this.currLap, y: dataPoint.totalPowerOut});
+      this.netPowerOutArray.push({x: this.currLap, y: dataPoint.netPowerOut});
+      this.distanceArray.push({x: this.currLap, y: dataPoint.distance});
+      this.amphoursArray.push({x: this.currLap, y: dataPoint.amphours});
+      this.averagePackCurrentArray.push({x: this.currLap, y: dataPoint.averagePackCurrent});
+      this.batterySecondsRemainingArray.push({x: this.currLap, y: dataPoint.batterySecondsRemaining});
+  }
+
+  // event Handler for radio button selection change
+  radioChangeHandler (event: any) {
+    this.selectedGraph = event.value;
+    console.log('Graph selection changed to ' + event.value);
+    this.updateChart();
+  }
+
+  convertToMinutes(timeString: string) {
+    const splitTime = timeString.split(':', 3)
+    return Number(splitTime[0]) * 60 + Number(splitTime[1]) + Number(splitTime[2]) / 60
+  }
+
+  updateChart() {
+    let arrayToDisplay: GraphDataPoint[];
+    let arrayToDisplayTwo: GraphDataPoint[];
+    let arrayToDisplayThree: GraphDataPoint[];
```

Comment:
Let's make a new struct/data type containing the `dataPoints`, `showInLegend`, and `legendText` information. Then, instead of 3 GraphDataPoint variables and an array of booleans, we can have an array of our new data type that we can work with instead.

Updating the inputs would look like (not exact syntax):

```
let myArray: myObject[]
if (this.selectedGraph === 'Time') {
      graphHeader = 'Lap Time';
      yLabel = 'Time (minutes)';
      myArray.push(new myObject(this.lapTimeArray, false,'') 
}
```

Then later:
```
for(let i = 0; i < myArray.length; i++)
{
this.chart.options.data[i].push(myArray[i])
}
```

Reply:
I'm not sure I understand exactly what you're trying to say. Are you essentially asking to wrap the 3 input arrays to the graph, showLegend, and legendText into one object so we can then loop instead of explicitly updating the inputs in lines 182 to 192?

If so, why?

Reply:
The main goal is to do away with the need to have stuff like `arrayToDisplay`, `arrayToDisplayTwo`, `arrayToDisplayThree`, etc.

Say in the future, we're requested to add new graph with more than 3 sets of data, we we'd have to:

- add a new `arrayToDisplayFour`, 
- add a new index to the current `showLegend` array
- propagate that to every if statement

Which would be pretty difficult to work with as it scales upwards. I'm thinking that if we can loop through instead, then it would make it a bit more maintainable later.

Reply:
sounds good